src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,99 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import classNames from 'classnames';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  addComment: (comment: CommentData) => void;
+  isCommentAdditionError: boolean;
+  isCommentAdditionLoading: boolean;
+};
+
+export const NewCommentForm: React.FC<Props> = React.memo(({
+  postId,
+  addComment,
+  isCommentAdditionError,
+  isCommentAdditionLoading,
+}) => {
+  const [userName, setUserName] = useState('');
+  const [userEmail, setUserEmail] = useState('');
+  const [userComment, setUserComment] = useState('');
+  const [isUserRequired, setIsUserRequired] = useState(false);
+  const [isUserEmailRequired, setIsUserEmailRequired] = useState(false);
+  const [isUserCommentRequired, setIsUserCommentRequired] = useState(false);
+
+  const handleSubmit = useCallback((event: React.FormEvent) => {
+    event.preventDefault();
```

Comment:
It's better to use a separate function for form validation and call it inside handleSubmit function. This will make the code more modular and maintainable.